src/fparser/two/tests/fortran2003/test_defined_unary_op_r703.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2019 Science and Technology Facilities Council
```

Comment:
I'm concerned that these tests are just a duplicate of those for the binary_op (since the rules for binary_op duplicate those for unary_op). We could use a fixture that specified the type of operator to test and then just have a single set of tests. However, this would then break our own rule for having a separate test file for each rule in the spec. Discuss :-)

Reply:
I don't really mind.

I went for duplication as it followed our "separate test file per rule" rule and it was easy to replicate. I don't think these particular tests will change very often from now on so I don't think there will be much of a code maintenance issue. It also happens to be the existing working solution so would involve me doing nothing if we agreed to stay as we are!

However, I am also not wed to the "separate test file per rule" if you think it makes more sense to join them. We could call the merged file something like "test_defined_any_op_r703_r723.py" and use a fixture as you suggest.

I'm happy for you to make the call.

Reply:
OK, I've done this. I couldn't find a way to get the class name from a class so have had to simplify the repr() comparisons in two of the tests.